### PR TITLE
Serialisation from Pickle to Protobuf

### PR DESCRIPTION
**Pull request by 5A11**
## Proposed changes

The generator creates the proto schema file; 
The generator creates the pbk_2 file; 
The serialisation method is changed from pickle to protobuf; 
The code is cleaned up
## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/protocols/generator.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 MESSAGE_FILE_NAME = "message.py"
 SERIALIZATION_FILE_NAME = "serialization.py"
 PROTOCOL_FILE_NAME = "protocol.yaml"
+PATH_TO_PROTO_SCHEMA_FILE = "packages.protocols."
```
Comment (marcofavorito) [inline aea/protocols/generator.py]: Is it used somewhere?

File: aea/protocols/generator.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
...
 
 BASIC_FIELDS_AND_TYPES = {
```
Comment (marcofavorito) [inline aea/protocols/generator.py]: Is it used somewhere?

File: aea/protocols/generator.py (hunk 1 above)
Comment (5A11) [reply aea/protocols/generator.py]: No. Deleted.

File: examples/protocol_specification_ex/sample.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ speech_acts:
     proposal: pt:optional[pt:dict[ct:IOTApp7, pt:bytes]]
     rounds: pt:set[pt:int]
     items: pt:list[ct:Unit]
+    conditions: pt:optional[pt:union[pt:str, pt:dict[pt:str,pt:int], pt:set[ct:DataModel], pt:dict[pt:bytes, pt:float]]]
```
Comment (Totoual) [inline examples/protocol_specification_ex/sample.yaml]: Is this possible to make it in the form of a dictionary in the future? It is kind of difficult
to read :)

Comment (5A11) [reply examples/protocol_specification_ex/sample.yaml]: That's just to test the functionality. Any developer who is creating a protocol specification is responsible for its complexity. 

The sample specification is just to show (and test) various possibilities.

**Pull request merged**